hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java

```diff
@@ -742,12 +744,27 @@ public void onConfigurationChange(Configuration newConf) {
       }
     }
 
-    ThroughputController old = this.compactionThroughputController;
-    if (old != null) {
-      old.stop("configuration change");
+    String newCompactClass = newConf.get(
+        CompactionThroughputControllerFactory.HBASE_THROUGHPUT_CONTROLLER_KEY);
+    String oldCompactClass = conf.get(
```

Comment:
get oldCompactClass  from compactionThroughputController? conf and newConf should ref to the same object.

Reply:
> get oldCompactClass from compactionThroughputController? conf and newConf should ref to the same object.

Thank you .I'll modify the code.